cura/PrinterOutputDevice.py

```diff
@@ -0,0 +1,303 @@
+    #   Note that this is a relative move. If you want to move the head to a specific position you can use
+    #   setHeadPosition
+    #   This function is "final" (do not re-implement)
+    #   /param x distance in x to move
+    #   /param y distance in y to move
+    #   /param z distance in z to move
+    #   /param speed Speed by which it needs to move (in mm/minute)
+    #   /sa _moveHead implementation function
+    @pyqtSlot("long", "long", "long")
+    @pyqtSlot("long", "long", "long", "long")
+    def moveHead(self, x = 0, y = 0, z = 0, speed = 3000):
+        self._moveHead(x, y, z, speed)
+
+    ##  Implementation function of moveHead.
+    #   /param x distance in x to move
+    #   /param y distance in y to move
+    #   /param z distance in z to move
+    #   /param speed Speed by which it needs to move (in mm/minute)
+    #   /sa moveHead
+    def _moveHead(self, x, y, z, speed):
+        Logger.log("w", "_moveHead is not implemented by this output device")
+
+    ##  Implementation function of setHeadPosition.
+    #   /param x new x location of the head.
+    #   /param y new y location of the head.
+    #   /param z new z location of the head.
+    #   /param speed Speed by which it needs to move (in mm/minute)
+    #   /sa setHeadPosition
+    def _setHeadPosition(self, x, y, z, speed):
+        Logger.log("w", "_setHeadPosition is not implemented by this output device")
+
+    ##  Implementation function of setHeadX.
+    #   /param x new x location of the head.
+    #   /param speed Speed by which it needs to move (in mm/minute)
+    #   /sa setHeadX
+    def _setHeadX(self, x, speed):
+        Logger.log("w", "_setHeadX is not implemented by this output device")
+
+    ##  Implementation function of setHeadY.
+    #   /param y new y location of the head.
+    #   /param speed Speed by which it needs to move (in mm/minute)
+    #   /sa _setHeadY
+    def _setHeadY(self, y, speed):
+        Logger.log("w", "_setHeadY is not implemented by this output device")
+
+    ##  Implementation function of setHeadZ.
+    #   /param z new z location of the head.
+    #   /param speed Speed by which it needs to move (in mm/minute)
+    #   /sa _setHeadZ
+    def _setHeadZ(self, z, speed):
+        Logger.log("w", "_setHeadZ is not implemented by this output device")
+
+    ##  Get the progress of any currently active process.
+    #   This function is "final" (do not re-implement)
+    #   /sa _getProgress
+    #   /returns float progress of the process. -1 indicates that there is no process.
+    @pyqtProperty(float, notify = progressChanged)
+    def progress(self):
+        return self._progress
+
+    ##  Set the progress of any currently active process
+    #   /param progress Progress of the process.
+    def setProgress(self, progress):
+        self._progress = progress
+        self.progressChanged.emit()
+
+##  The current processing state of the backend.
+class ConnectionState(IntEnum):
+    CLOSED = 0
```

Comment:
We never formally agreed on it iirc and I have always been against it. As I said, all other enums/constant values in Uranium and Cura use upper camel case as far as I know, so I would at least prefer to keep that consistent.